interval/search_tree.go

```diff
@@ -178,3 +221,43 @@ func (st *MultiValueSearchTree[V, T]) IsEmpty() bool {
 
 	return st.root == nil
 }
+
+// GobEncode encodes the tree (compatible with encoding/gob).
```

Comment:
What do you think of improving the comment by adding link to the `encoding/gob`, so readers can go straight to it from this package documentation?
```suggestion
// GobEncode encodes the tree (compatible with [encoding/gob]).
```

See the [links section](https://tip.golang.org/doc/comment#links) of Golang comments guide for more on this.

Reply:
Done: [Add links to [encoding/gob]](https://github.com/rdleal/intervalst/pull/9/commits/fa748e2e243ac7e6142d7ebb8b43f9dc46f8dd71)